config/config_params.go

```diff
@@ -101,6 +101,56 @@ func (source Source) Local() bool {
 	}
 }
 
+// Provider represents a particular provider or flavor of Kubernetes.
+type Provider uint8
+
+const (
+	ProviderNone Provider = iota + 1
```

Comment:
Go idiom is to make the "zero" value useful.  Why not have `ProviderNone==0` ?

Reply:
I can't remember why I thought 0 would be a problem, tried it now and seems fine!